src/Symfony/Component/Console/Helper/ProgressBar.php

```diff
@@ -600,4 +614,23 @@ private static function initFormats()
             'debug_nomax' => ' %current% [%bar%] %elapsed:6s% %memory:6s%',
         );
     }
+
+    /**
+     * @param string $line
+     * @return bool
+     */
+    private function adjustBarWidthToWindowWidth($line)
+    {
+        $lineLength = Helper::strlenWithoutDecoration($this->output->getFormatter(), $line);
+        $windowWidth = $this->terminalDimensionsProvider->getTerminalWidth();
+        if ($lineLength > $windowWidth) {
+            $barWidthDelta = $lineLength - $windowWidth;
+            $newBarWidth = $this->barWidth - $barWidthDelta - 20;
```

Comment:
It shall be calculated as "terminal width - everything except bar width (number, %, memory etc.)"

But first, I need to find way to test this properly.